src/accelerate/state.py

```diff
@@ -89,6 +89,11 @@ def __init__(
             dynamo_backend = (
                 parse_choice_from_env("ACCELERATE_DYNAMO_BACKEND", "no") if dynamo_backend is None else dynamo_backend
             )
+            env_device = os.environ.get("ACCELERATE_TORCH_DEVICE", None)
+            if env_device:
+                self.device = torch.device(env_device)
+            else:
+                self.device = None
```

Comment:
Let's use Sylvains suggestion of:

```suggestion
            self.device = torch.device(env_device) if env_device is not None else None
```

And then in all the checks do `if self.device is None: ...`